group_project_v2/stage.py

```diff
@@ -0,0 +1,625 @@
+            for peer_review_item in review_items
+            if peer_review_item['reviewer'] == self.anonymous_student_id
+        }
+
+        for item in items_to_grade:
+            for question in review_questions:
+                key = make_key(item["id"], question.question_id)
+                if my_feedback.get(key, None) in (None, ''):
+                    return False
+
+        return True
+
+    def _pivot_feedback(self, feedback):  # pylint: disable=no-self-use
+        """
+        Pivots the feedback to show question -> answer
+        """
+        return {pi['question']: pi['answer'] for pi in feedback}
+
+
+class PeerReviewStage(ReviewBaseStage, WorkgroupAwareXBlockMixin):
+    STAGE_CONTENT_TEMPLATE = 'templates/html/stages/peer_review.html'
+    CATEGORY = 'group-project-v2-stage-peer-review'
+
+    @property
+    def allowed_nested_blocks(self):
+        blocks = super(PeerReviewStage, self).allowed_nested_blocks
+        blocks.update(OrderedDict([
+            (PeerSelectorXBlock.CATEGORY, _(u"Teammate selector"))
+        ]))
+        return blocks
+
+    def is_review_complete(self):
+        peers_to_review = [user for user in self.workgroup["users"] if user["id"] != self.user_id]
+        peer_review_items = project_api.get_peer_review_items_for_group(self.workgroup['id'], self.content_id)
+
+        return self._check_review_complete(peers_to_review, self.required_questions, peer_review_items, "user")
+
+    def validate(self):
+        violations = super(PeerReviewStage, self).validate()
+
+        # Technically, nothing prevents us from allowing graded peer review questions. The only reason why
+        # they are considered not supported is that GroupActivityXBlock.calculate_grade does not
+        # take them into account.
+        if self.grade_questions:
+            violations.add(ValidationMessage(
+                ValidationMessage.ERROR,
+                _(u"Grade questions are not supported for {class_name} '{stage_title}'").format(
+                    class_name=self.__class__.__name__, stage_title=self.display_name
+                )
+            ))
+
+        return violations
+
+    @XBlock.handler
+    def load_peer_feedback(self, request, suffix=''):  # pylint: disable=unused-argument
+
+        peer_id = request.GET["peer_id"]
+        feedback = project_api.get_peer_review_items(
+            self.anonymous_student_id,
+            peer_id,
+            self.workgroup['id'],
+            self.content_id,
+        )
+
+        results = self._pivot_feedback(feedback)
+
+        return webob.response.Response(body=json.dumps(results))
+
+    @XBlock.json_handler
+    def submit_peer_feedback(self, submissions, suffix=''):  # pylint: disable=unused-argument
+        try:
+            peer_id = submissions["review_subject_id"]
+            del submissions["review_subject_id"]
+
+            # Then something like this needs to happen
```

Comment:
Something like this needs to happen? Is this not representative of final functionality, then?